enterprise/server/backends/distributed/distributed.go

```diff
@@ -995,6 +996,10 @@ func (c *Cache) Reader(ctx context.Context, d *repb.Digest, offset, limit int64)
 	return c.distributedReader(ctx, d, offset, limit)
 }
 
-func (c *Cache) Writer(ctx context.Context, d *repb.Digest) (io.WriteCloser, error) {
-	return c.multiWriter(ctx, d)
+func (c *Cache) Writer(ctx context.Context, d *repb.Digest) (interfaces.CommittedWriteCloser, error) {
+	mwc, err := c.multiWriter(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.AutoUpgradeCloser(mwc), nil
```

Comment:
note to self: should be able to remove this and just return an underlying cwc here